core/rawdb/chain_freezer.go

```diff
@@ -114,60 +158,39 @@ func (f *chainFreezer) freeze(db ethdb.KeyValueStore) {
 				return
 			}
 		}
-		// Retrieve the freezing threshold.
-		hash := ReadHeadBlockHash(nfdb)
-		if hash == (common.Hash{}) {
-			log.Debug("Current full block hash unavailable") // new chain, empty database
+		number, err := f.freezeThreshold(nfdb)
+		if err != nil {
 			backoff = true
+			log.Debug("Current full block not old enough to freeze")
```

Comment:
Maybe add the `"err", err` as a context to the debug log. I know we only have 1 possible error return now, but it might save us some brain cells later if we ever expand or rework the errors.